### ISSUE TITLE
Espruino Hub Home Assistant Add on creation?

### ISSUE THREAD

**Issue by bouton** [werty1st/EspruinoHub-docker#1]
Hello
I currently run espruino hub on a raspberry pi zero w, while running home assistant OS on a raspberry pi 3B+ (which gets installed as a docker). I found you on the espruino hub git issues.

I use espruino hub to translate by puck.js bluetooth advertising via button pushes into mqtt, which I use to control my sonos devices.

I was looking to be able to run espruino hub on the pi as well but am unfamiliar with docker and all the ins and outs. Then I found a page on developing add-ons for home assistant. https://developers.home-assistant.io/docs/add-ons/

This wouldn't be something you could do, would it?
If not, could you point me in the right direction.

I am thinking otherwise install a home assistant add on "portainer" and add your docker hub files into that.  Rather skip the additional step though.

Thanks

**Comment by werty1st** [comment]
Hi, I'm using my PuckJS together with https://esphome.io/ and Home Assistent.

https://user-images.githubusercontent.com/4736160/162289597-1f23b2b7-ab70-4261-8b84-1186adbcc900.mp4

The Code for the ESPHome device:
```yaml
esphome:
  name: m5c3-blue
  platformio_options:
    platform_packages:
      - framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#2.0.1
    platform:
      - https://github.com/platformio/platform-espressif32.git#feature/arduino-upstream

esp32:
  board: esp32-c3-devkitm-1
  framework:
    type: arduino

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

web_server:

esp32_ble_tracker:
  on_ble_service_data_advertise:
    - mac_address: ff:aa:bb:cc:dd:ee # Puck's Mac [NRF.getAddress()]
      service_uuid: 180F
      then:
        - lambda: 'id(puck2batlvl).publish_state(x[0]);'
        
  on_ble_manufacturer_data_advertise:
    - mac_address: ff:aa:bb:cc:dd:ee # Puck's Mac [NRF.getAddress()]
      manufacturer_id: 0590
      then:
        - lambda: |-
            ESP_LOGD("puck2switch", "int: %i", x[0]);
            if (x[0] > 0) {
              // Switch is ON, do something here
              id(puck2switch).publish_state(false);
            } else {
              // Switch is OFF, do something else here
              id(puck2switch).publish_state(true);
            }
            
      
binary_sensor:
  - platform: template
    name: "Puck2 BLE Switch"
    id: puck2switch


sensor:
  - platform: template
    name: "Puck2 Battery Level"
    id: puck2batlvl
    icon: 'mdi:battery'
    unit_of_measurement: '%'
```

PuckJS' code:

```javascript
var toggel = false;

function advertise() {

  NRF.setAdvertising(
    [
      {0x180F : [Puck.getBatteryPercentage()]}
    ],
    {
      manufacturer: 0x0590,
      manufacturerData:[toggel = !toggel],
      discoverable: true,
      connectable: false,
      scannable: false,
      name: "Puck2",
      showName: true
    }
  );
  if (toggel) {
    digitalPulse(LED1, true, 150);
  } else {
    digitalPulse(LED2, true, 150);
  }
}

setWatch(advertise, BTN, {edge:"rising", repeat:1, debounce:20});

```

So in don't think you need any new addon.

---

I have three M5Stamp C3 flashed with the ESPHome firmware to act as BLE to Wifi/Home Assistant Gateway in different rooms.

![IMG_3224](https://user-images.githubusercontent.com/4736160/162293828-a03912c2-9e45-4086-918a-573df5673aaf.gif)

But for my use case this is not an ideal solution. That's way I will also try this software: https://github.com/shmuelzon/esp32-ble2mqtt

**Comment by bouton** [comment]
Thanks for the great write up. I gave it try. and worked great - just a hair slower than the espruino hub.
I have written up what I did originally with the hub.
https://community.home-assistant.io/t/control-sonos-player-with-puck-js-hardware-button/391110

I was just trying to use one less piece of hardware and get espruino hub as a container on the same raspberry pi as my home assistant.

Thanks again
ps you might want to check out espresense for room sensing - works great and works with a puck or bangle
